### DISCUSSION TITLE
여기서 continuous는 어떤걸 의미하는걸까요?

### DISCUSSION THREAD

**Discussion by arabae** [bcaitech1/p4-dkt-freshtomato#25] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/46676700/119666881-bbb8d800-be70-11eb-842e-35222ba33b45.png)

이전에 추가해봤던 **User별 TestID에 따른 정답률 feature**는 categorical로 이해해서 우리가 기존에 사용했던 feature 3개와 동일한 작업을 거치도록 했는데,  
embedding layer를 거치고 이게 linaer에 들어가는 방식!

1. continuous한 feature는 어떤게 있는지 궁금하고,   
2. 내가 추가한 feature가 categorical이 맞는지? 혹은 내가 잘 못 이해한건지 궁금해!  

---
➕  
### **추가한 feature를 포함한 dataFrame 일부**
![image](https://user-images.githubusercontent.com/46676700/119667571-5fa28380-be71-11eb-868e-b6a60a0ab57b.png)

혹시 몰라서 추가된 feature를 포함하는 df 사진도 같이 첨부해~

**Comment by MoonJaeHoon** [comment, marked as answer]
추가한 feautre가 CorrectRatio인거지?
내가 보기에 이건 continuous feature같은뎁

음.. 강의 나중에 한번 다시 보고 답변달게연..

**Reply by arabae** [reply]
오호 그러면 각 0번 user의 경우 50 row 특징이 있다고 생각하면, 그냥 correctRatio는 linear layer에 (batch, 50, 1) 이렇게 들어가는건가..?

**Reply by MignonDeveloper** [reply]
오오!! 나도 continuous 같은데, 그러면 누나가 만들어준 feature는 저렇게 embedding 과정을 거치지 않고 바로 linear layer로 통과시켜줘야 하는걸까??

**Reply by arabae** [reply]
처음에 continuous같았는데 저 linear에 바로 들어가는 부분이 이상해서 일단 embedding을 거친 뒤 해봤거든! 근데 다들 이야기 들어보고 다시 바로 linear에 들어가는 방향으로 바꿔봤어!!
우리 code를 기준으로 보면 linear가 embedding을 concat한 뒤에 수행돼서, (batch, max_seq_len, 1) shape인 correctRatio도 일단 concat할 때 같이 합쳐서 수행하도록했어 😊

**Reply by arabae** [reply]
내가 여기서 고민한거였는데 이전의 feature 3개는 embedding layer를 거치기때문에 (batch, max_seq_len, hidden/3) shape인데 새로 추가하는 feature는 (batch, max_seq_len, 1)이다 보니까 너무 적은 부분을 차지하는게 아닌가 싶었거든 그러면 (batch, max_seq_len, 1)을 먼저 linear layer를 사용해서 feature dimension을 맞춰준 뒤에 합쳐서 사용하는게 맞는 방법일까?

**Reply by MignonDeveloper** [reply]
응응 누나가 말한대루 concat하기전에 linear layer를 통과시켜서 continuous feature도 feature dimension을 맞춰주면 좋을 것 같아! 오늘 강의에서 나온것 처럼! 그다음에 comb_proj layer 통과시키면 될것같아!

**Reply by arabae** [reply]
넹넹!!! 🤗🤗🤗
